src/group/libp2p_group_relcast_server.erl

```diff
@@ -247,8 +245,27 @@ handle_cast({handle_data, Index, Msg, Seq}, State=#state{self_index=_SelfIndex})
                     _ ->
                         false
                 end,
-    case relcast:deliver(Seq, Msg, Index, State#state.store) of
-        full ->
+    Res =
+        lists:foldl(
+          fun(M, {full, RC, Acc}) ->
+                  {full, RC, [M | Acc]};
+             ({Seq, Msg}, RC) ->
+                  case relcast:deliver(Seq, Msg, Index, RC) of
+                      full ->
+                          {full, RC, [{Seq, Msg}]};
+
+                  {ok, NewRC} ->
+                      NewRC;
+                  %% just keep looping, I guess, it kind of doesn't matter?
+                  {stop, Timeout, NewRC} ->
+                      erlang:send_after(Timeout, self(), force_close),
```

Comment:
should we limit the number of timers we set here?

Reply:
we could, but subsequent timers won't have any further effect since after the first one lands, the process will be dead and the message will be dropped by ERTS.

Reply:
Yeah.